fog/api/proto/ingest.proto

```diff
@@ -95,3 +101,48 @@ message SyncKeysFromRemoteRequest {
     /// synced.
     string peer_uri = 1;
 }
+
+message GetIngressKeyRecordsRequest {
+    /// Ingress keys are "started" at certain blocks. Only ingress keys that
+    /// are "started"  at this block index will be included in the response.
+    uint64 start_block_at_least = 1;
+
+    /// If true the response will include ingress keys that have been lost.
+    bool should_include_lost_keys = 2;
+
+    /// If true the response will include ingress keys that have been retired.
+    bool should_include_retired_keys = 3;
+}
+
+message GetIngressKeyRecordsResponse {
+    /// The records that filtered according to the GetIngressKeyRecordsRequest.
+    repeated IngressPublicKeyRecord records = 1;
+}
+
+/// Corresponds to the IngressPublicKeyRecord struct found in
+/// mc-fog-recovery-db-iface.
+message IngressPublicKeyRecord {
```

Comment:
This message is derived from the IngressPublicKeyRecord defined [here](https://github.com/mobilecoinfoundation/mobilecoin/blob/bcb3e26cc85afd3a250f2109e9cd388c0e5c814a/fog/recovery_db_iface/src/types.rs). I opted to recreate the definition here rather add prost tags to that struct because of the comment at the top of that file that reads: "These are not user-facing, the user facing versions are in fog-types crate."

However, let me know if you'd like me to define this message in a different location (fog-types) or in a different way (using prost etc.)

Reply:
i think this is the right place for this, thank you!